packages/rpc-graphql/src/loaders/loader.ts

```diff
@@ -7,7 +7,8 @@ import { Commitment, Slot } from '@solana/rpc-types';
 import stringify from 'json-stable-stringify';
 
 export type LoadFn<TArgs, T> = (args: TArgs) => Promise<T>;
-export type Loader<TArgs, T> = { load: LoadFn<TArgs, T> };
+export type LoadManyFn<TArgs, T> = (args: TArgs[]) => Promise<(Promise<T> | Error)[]>;
```

Comment:
It's super subtle, but no. The _reason_ that method is `Promise<Promise<T | Error>[]>` is because the method itself _returns_ a bunch of promises, but has _async_ code in the `finally` block, like this:

https://www.typescriptlang.org/play?#code/BQegVABAhgzgngOwMYTCCwCUEC8A+CAbwCgIyIAXAJziNPIaoFMKBXKhCAbXob4gAKVAPYBbAJYwmAOmYxhAGwBuTYAEZMAGl79yQsZJlzFK4ACYtO3YJESpspvOWqAzJesBdANw6AvhAAzcQQoBQVaEmsoAHcocQobA3tjZ2AAcgDhYTTMHwZfYl9sGCgKSSDHDGx8RLsmAB59OvqEVlEAIyYqPC4PPCA

FWIW I don't think my [original take](https://github.com/solana-labs/solana-web3.js/pull/2063/files) had `await` in the `finally` block and yours doesn't have to either. All you want to do is make sure all the requests are fired off. You don't actually need to `await Promise.all()` I don't think.

Reply:
I've corrected this!